src/main/java/Duck.java

```diff
@@ -0,0 +1,161 @@
+
+        // Start scanner
+        Scanner sc = new Scanner(System.in);
+
+        // Print greeting
+        printAsResponse(GREETING);
+
+        // User input loop
+        while (true) {
+            String line = sc.nextLine();
+            Parser lineBuffer = new Parser(line);
+
+            // "bye" and "list" will only work if they are the only word in the line
+            // e.g. "bye bye" would not work
+            if (Command.BYE.equalsName(line)) {
+                printAsResponse(GOODBYE);
+                break;
+            } else if (Command.LIST.equalsName(line)) {
+                String response = "Here are the tasks in your list:\n"
+                        + TASKS.toString();
+                printAsResponse(response);
+            } else {
+                String command = lineBuffer.getWord();
+
+                // "mark" / "unmark" commands
+                if (Command.FIND.equalsName(command)) {
+                    String pattern = lineBuffer.getRemainingLine();
+                    TaskList filteredTasks = TASKS.filterTasksByPattern(pattern);
+                    String response = "Here are the tasks in your list:\n"
+                            + filteredTasks.toString();
+                    printAsResponse(response);
+                } else if (Command.MARK.equalsName(command)) {
+                    int taskLabel = lineBuffer.getInt();
+                    Task task = TASKS.getItem(taskLabel);
+                    task.markAsDone();
+                    TASKS.updateFileWithTaskList();
+
+                    String response = "Nice! I've marked this task as done:\n"
+                            + indentText(task.toString(), 2);
+                    printAsResponse(response);
+                } else if (Command.UNMARK.equalsName(command)) {
+                    int taskLabel = lineBuffer.getInt();
+                    Task task = TASKS.getItem(taskLabel);
+                    task.markAsNotDone();
+                    TASKS.updateFileWithTaskList();
+
+                    String response = "OK, I've marked this task as not done yet:\n"
+                            + indentText(task.toString(), 2);
+                    printAsResponse(response);
+                } else if (Command.DELETE.equalsName(command)) {
+                    // Delete
+                    int taskLabel = lineBuffer.getInt();
+                    Task task = TASKS.getItem(taskLabel);
+                    TASKS.removeItem(taskLabel);
+
+                    String response = "Noted. I've removed this task:\n"
+                            + indentText(task.toString(), 2) + "\n"
+                            + String.format("Now you have %s tasks in the list.", TASKS.getTaskCount());
+                    printAsResponse(response);
+                } else if (Command.TODO.equalsName(command)) {
+                    // Tasks
+                    // TODO: Reduce duplicate code
+
+                    String taskPart = lineBuffer.getRemainingLine();
```

Comment:
Do you think the name taskPart could cause some confusion regarding what 'part' is being referred to? Maybe something like taskDetails might be more relevant in this context.